File: backend/models/saved_tab.go
package models

import "time"

type SavedTab struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	UserID      uint      `gorm:"not null;index" json:"user_id"`
	TabID       string    `gorm:"not null" json:"tab_id"`
	Name        string    `json:"name"`
	Method      string    `json:"method"`
	URL         string    `json:"url"`
	Headers     string    `gorm:"type:text" json:"headers"`      // JSON string
	Body        string    `gorm:"type:text" json:"body"`
	QueryParams string    `gorm:"type:text" json:"query_params"` // JSON string
	IsActive    bool      `json:"is_active"`
	SortOrder   int       `json:"sort_order"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: desktop/src/main/java/uz/postbaby/desktop/model/SavedTab.java
package uz.postbaby.desktop.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;

import java.util.LinkedHashMap;
import java.util.Map;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class SavedTab {
    public String tab_id;
    public String name;
    public String method;
    public String url;
    public Map<String, String> headers = new LinkedHashMap<>();
    public String body;
    public Map<String, String> query_params = new LinkedHashMap<>();
    public boolean is_active;
    public int sort_order;

    public Authorization authorization;

    public SavedTab() {
    }
}
